crates/sui-framework/packages/bridge/tests/bridge_setup.move

```diff
@@ -0,0 +1,695 @@
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.btc_coins.split(amount, ctx)
+    }
+
+    public fun get_eth(env: &mut BridgeEnv, amount: u64): Coin<ETH> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.eth_coins.split(amount, ctx)
+    }
+
+    public fun get_usdc(env: &mut BridgeEnv, amount: u64): Coin<USDC> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.usdc_coins.split(amount, ctx)
+    }
+
+    public fun get_usdt(env: &mut BridgeEnv, amount: u64): Coin<USDT> {
+        let scenario = &mut env.scenario;
+        let ctx = scenario.ctx();
+        env.vault.usdt_coins.split(amount, ctx)
+    }
+
+    //
+    // Bridge commands
+    //
+
+    // Register new tokens
+    public fun register_foreign_token<T>(
+        env: &mut BridgeEnv,
+        treasury_cap: TreasuryCap<T>,
+        upgrade_cap: UpgradeCap,
+        metadata: CoinMetadata<T>,
+        sender: address,
+    ) {
+        let scenario = env.scenario();
+        scenario.next_tx(sender);
+        let mut bridge = scenario.take_shared<Bridge>();
+        bridge.register_foreign_token<T>(treasury_cap, upgrade_cap, &metadata);
+        test_scenario::return_shared(bridge);
+        destroy(metadata);
+    }
+
+    // Freeze the bridge
+    public fun freeze_bridge(env: &mut BridgeEnv, sender: address, error: u64) {
+        let scenario = env.scenario();
+        scenario.next_tx(sender);
+        let mut bridge = scenario.take_shared<Bridge>();
+        let inner = bridge.test_load_inner_mut();
+        let msg = message::create_emergency_op_message(env.chain_id, 0, emergency_op_pause());
+        let payload = msg.extract_emergency_op_payload();
+        inner.test_execute_emergency_op(payload);
+        inner.assert_paused(error);
+        test_scenario::return_shared(bridge);
+    }
+
+    // Unfreeze the bridge
+    public fun unfreeze_bridge(env: &mut BridgeEnv, sender: address, error: u64) {
+        let scenario = env.scenario();
+        scenario.next_tx(sender);
+        let mut bridge = scenario.take_shared<Bridge>();
+        let inner = bridge.test_load_inner_mut();
+        let msg = message::create_emergency_op_message(env.chain_id, 1, emergency_op_unpause());
+        let payload = msg.extract_emergency_op_payload();
+        inner.test_execute_emergency_op(payload);
+        inner.assert_not_paused(error);
+        test_scenario::return_shared(bridge);
+    }
+
+    public fun send_token<T>(
+        env: &mut BridgeEnv,
+        target_chain_id: u8,
+        eth_address: vector<u8>,
+        coin: Coin<T>,
+    ) {
+        let scenario = env.scenario();
+        scenario.next_tx(@0xAAAA);
+        let mut bridge = scenario.take_shared<Bridge>();
+        bridge.send_token(target_chain_id, eth_address, coin, scenario.ctx());
+        test_scenario::return_shared(bridge);
```

Comment:
we can check a few things here:
1. assert the burn happened
2. token_transfer_records has a new entry with correct metadata